### PR TITLE
Implement audio for macOS

### PR DESCRIPTION
**Pull request by IntelOrca**
#221 was split off for macOS due to SDL2_mixer difficulties. This PR gets audio working for macOS.

Will rebase on top of master when #221 is merged.

### TODO
* [x] Resolve macOS SDL2-mixer dependency

### CODE REVIEW

File: src/openloco/audio/audio.cpp [hunk 1]
```diff
@@ -426,6 +426,7 @@ namespace openloco::audio
     const char* get_current_device_name()
     {
         auto index = get_current_device();
+#ifdef __HAS_DEFAULT_DEVICE__
```
Comment (AaronVanGeffen) [inline src/openloco/audio/audio.cpp]: Rebase error

File: .travis.yml [hunk 2]
```diff
@@ -12,19 +12,19 @@ matrix:
...
           env:
-            - OPENLOCO_CMAKE_OPTS="-G Ninja -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=release -DSDL2_DIR=/usr/lib/i386-linux-gnu/cmake/SDL2 -Dyaml-cpp_DIR=/usr/lib/i386-linux-gnu/cmake/yaml-cpp"
+            - OPENLOCO_CMAKE_OPTS="-G Ninja -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=release -DSDL2_DIR=/usr/lib/i386-linux-gnu/cmake/SDL2 -DSDL2MIXERDIR=/usr/lib/i386-linux-gnu -Dyaml-cpp_DIR=/usr/lib/i386-linux-gnu/cmake/yaml-cpp"
             - DOCKERIMG=ubuntu-i686
         - os: linux
           services:
             - docker
           env:
-            - OPENLOCO_CMAKE_OPTS="-G Ninja -DCMAKE_TOOLCHAIN_FILE=../CMakeLists_mingw.txt -DCMAKE_BUILD_TYPE=release -DBOOST_ROOT=/usr/i686-w64-mingw32/sys-root/mingw/ -DSDL2_DIR=/usr/lib/i386-linux-gnu/cmake/SDL2 -Dyaml-cpp_DIR=/usr/i686-w64-mingw32/sys-root/mingw/CMake/"
+            - OPENLOCO_CMAKE_OPTS="-G Ninja -DCMAKE_TOOLCHAIN_FILE=../CMakeLists_mingw.txt -DCMAKE_BUILD_TYPE=release -DBOOST_ROOT=/usr/i686-w64-mingw32/sys-root/mingw/ -DSDL2_DIR=/usr/lib/i386-linux-gnu/cmake/SDL2 -DSDL2MIXERDIR=/usr/i686-w64-mingw32/sys-root/mingw -Dyaml-cpp_DIR=/usr/i686-w64-mingw32/sys-root/mingw/CMake/"
```
Comment (janisozaur) [inline .travis.yml]: I used `-DSDL2_MIXER_PATH=/usr/i686-w64-mingw32/sys-root/mingw/`

**Comment by janisozaur** [comment]
I can confirm audio is working on Linux

**Comment by AaronVanGeffen** [comment]
It's taken some time, but I've finally managed to create an archive of OpenLoco's dependencies with `vcpkg`, and gotten Travis to use this archive whilst compiling and linking macOS builds.

I've not been able to get `vcpkg`-generated Boost libraries to link properly, however. Therefore, I've left the `brew` approach to boost as-is, for now.

**Comment by ras0219-msft** [comment]
@AaronVanGeffen Can you link to some logs where you were having trouble with boost? Or even better, open an issue on https://github.com/Microsoft/vcpkg with them and a repro :D

**Comment by AaronVanGeffen** [comment]
@ras0219-msft Thank you for reaching out once more! I've filed https://github.com/Microsoft/vcpkg/issues/4437 to discuss the Boost issue.

For now, with @marijnvdwerf's blessing, I'll merge this, so we finally have audio on macOS at least. :)

**Pull request merged**